testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/AbstractTestHiveViews.java

```diff
@@ -362,6 +366,72 @@ public void testNestedGroupBy()
                         row(4, 1)));
     }
 
+    @Test(groups = HIVE_VIEWS)
+    public void testUnionAllViews()
+    {
+        onHive().executeQuery("DROP TABLE IF EXISTS union_helper");
+        onHive().executeQuery("CREATE TABLE union_helper (\n"
+                + "r_regionkey BIGINT,\n"
+                + "r_name VARCHAR(25),\n"
+                + "r_comment VARCHAR(152)\n"
+                + ")");
+        onHive().executeQuery("INSERT INTO union_helper\n"
+                + "SELECT r_regionkey % 3, r_name, r_comment FROM region");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS union_all_view");
+        onHive().executeQuery("CREATE VIEW union_all_view AS\n"
+                + "SELECT r_regionkey FROM region\n"
+                + "UNION ALL\n"
+                + "SELECT r_regionkey FROM union_helper\n");
+
+        assertThat(query("SELECT r_regionkey FROM union_all_view"))
+                // Copy the keys 5 times because there are 5 nations per region
+                .containsOnly(
+                        // base rows
+                        row(0),
+                        row(1),
+                        row(2),
+                        row(3),
+                        row(4),
+                        // mod 3
+                        row(0),
+                        row(1),
+                        row(2),
+                        row(0),
+                        row(1));
+    }
+
+    @Test(groups = {HIVE_VIEWS, SKIP_ON_CDH})
```

Comment:
Add a comment why skip.

Also I would prefer to skip based on actual hive version. From the test code. Can't we have it?

Reply:
Updated to skip based on Hive version, with the skip message serving as the explanation instead of a comment.